src/Compilers/Core/Portable/Text/SourceText.cs

```diff
@@ -801,7 +801,7 @@ private int[] ParseLineStarts()
                 var index = 0;
                 if (lastWasCR)
                 {
-                    if (buffer.Length > 0 && buffer[0] == '\n')
+                    if (length > 0 && buffer[0] == '\n')
```

Comment:
The codepath is hit very often, but condition is false only when \r is last in the buffer. That is not common.
 In other cases length > 0 and buffer is never empty.